scripts/spokepool.ts

```diff
@@ -92,6 +99,22 @@ function getTokenAddress(symbol: string, chainId: number): ERC20 {
   return { address: addresses[chainId], decimals, symbol };
 }
 
+/**
+ * Resolves an ERC20 type from an address and chain ID.
+ * @param address The address of the token to resolve.
+ * @param chainId The chain ID to resolve the token on.
+ * @returns The ERC20 type of the token.
+ */
+function getTokenSymbol(address: string, chainId: number): ERC20 {
+  const token = Object.values(contracts.TOKEN_SYMBOLS_MAP).find(({ addresses }) => addresses[chainId] === address);
+  if (token === undefined) {
```

Comment:
We probably can't rely on the user supplying an address in checksummed-case:

```suggestion
  const _address = address.toLowerCase();
  const token = Object.values(contracts.TOKEN_SYMBOLS_MAP).find(({ addresses }) => addresses[chainId] === _address);
  if (token === undefined) {
```